plugin.video.popcorntv/default.py

```diff
@@ -0,0 +1,98 @@
+import os
+import sys
+import xbmc
+import xbmcgui
+import xbmcplugin
+import xbmcaddon
+import urllib
+import urlparse
+from resources.lib.popcorntv import PopcornTV
+
+# plugin constants
+__plugin__ = "plugin.video.popcorntv"
+__author__ = "Nightflyer"
+
+Addon = xbmcaddon.Addon(id=__plugin__)
+
+# plugin handle
+handle = int(sys.argv[1])
+
+# utility functions
+def parameters_string_to_dict(parameters):
+    ''' Convert parameters encoded in a URL to a dict. '''
+    paramDict = dict(urlparse.parse_qsl(parameters[1:]))
+    return paramDict
+ 
+def addDirectoryItem(parameters, li):
+    url = sys.argv[0] + '?' + urllib.urlencode(parameters)
+    return xbmcplugin.addDirectoryItem(handle=handle, url=url, 
+        listitem=li, isFolder=True)
+
+def addLinkItem(parameters, li):
+    li.setProperty('IsPlayable', 'true')
+    url = sys.argv[0] + '?' + urllib.urlencode(parameters)
+    return xbmcplugin.addDirectoryItem(handle=handle, url=url, 
+        listitem=li, isFolder=False)
+
+# UI builder functions
+def show_root_folder():
+    popcorntv = PopcornTV()
+    items = popcorntv.getCategories()
+
+    for item in items:
+        liStyle=xbmcgui.ListItem(item["title"])
+        addDirectoryItem({"mode": "folder", "url": item["url"]}, liStyle)
+    xbmcplugin.endOfDirectory(handle=handle, succeeded=True)
+
+def show_category_folder(url):
+    xbmc.log("Category URL: " + url)
+    popcorntv = PopcornTV()
+    items = popcorntv.getSubCategories(url)
+
+    for item in items:
+        liStyle=xbmcgui.ListItem(item["title"])
+        addDirectoryItem({"mode": "list", "url": item["url"]}, liStyle)
+    xbmcplugin.addSortMethod(handle, xbmcplugin.SORT_METHOD_LABEL)
+    xbmcplugin.endOfDirectory(handle=handle, succeeded=True)
+    
+def show_video_files(url):
+    xbmc.log("Subcategory URL: " + url)
+    popcorntv = PopcornTV()
+    page = popcorntv.getVideoBySubCategories(url)
+    
+    for video in page["videoList"]:
+        liStyle=xbmcgui.ListItem(video["title"], thumbnailImage=video["thumb"])
```

Comment:
thumbnailImage kwarg is deprecated please use liStyle.setArt({"thumb":xxxx, "icon": xxx, "fanart": xxxx, etc})